crates/gitbutler-tauri/src/open.rs

```diff
@@ -1,24 +1,28 @@
 use crate::error::Error;
+use anyhow::anyhow;
+use regex::Regex;
 use tracing::instrument;
 
-pub fn open_that(path: String) {
-    // `open` required to run in separate thread, to avoid blocking on some
-    // platforms (eg Fedora38 blocks)
+pub fn open_that(path: String) -> Result<(), Error> {
+    let re = Regex::new(r"^((https://)|(http://)|(mailto:)|(vscode://)|(vscodium://)).+").unwrap();
```

Comment:
I'd use the `url` crate here, parse the URL, and check what protocol/schema it's using. Then a match can be used which I'd find more maintainable than a long one-line regex.
Alternatively, I think `regex` might support multi-line regex.

Reply:
Yeah good idea. This is all still a bit of a PoC, but I'll definitely implement that if we decide to go ahead with this :ok_hand:

Reply:
Okay so I'm going back through your comments now, and regarding this one - these are all either actual URL protocols (`http`,`https`,`mailto`) or custom protocols that should be handled like URLs by the OS, no? 

For example, these can all be passed on to `xdg-open` on Linux or `open` on MacOS and the OS then needs to query its list of protocol handlers for the right app, for example for `vscodium://`.

I think we'd be doing unnecessary work trying to find the right binary to pass it to at this point

Reply:
If you are saying that it should be fine to let `open()` handle the URL entirely without pre-filtering it, then I agree.

Reply:
Isn't this regex something that should be removed entirely, or did I misunderstand something?
It's surprising that this was deemed necessary as we control all the URLs that are passed.

Reply:
Hmm yeah so we had it previously as an additional layer of protection in the `shell-open` configuration. 

That reminds me, we still have 1 place actually that users can pass links and that's via PR descriptions / into our markdown renderer. It's currenlty still just creating an anchor tag, which won't work on linux/appimage

Reply:
Thanks for elaborating! 
Alright, then it seems like the check should remain. Personally I find regex too flimsy and would definitely prefer 'harder' parsers here, as provided by the `url` crate.